Add shared prettier config and disable formatting rules

craigayre opened the pull request:
### Summary of Changes 📋

- Add prettier dependency and rules to be used across projects.
- Disable formatting linting rules

darkmavis1980 commented:
I think you should add prettier as a dependency, so we install it here

david-mooney reviewed (approved):
**Question** - are we separating the eslint major version update? Or do we want to sync that with these prettier changes?

darkmavis1980 commented:
As this version will bring breaking changes once implemented, I would recommend to bump the version to a major

craigayre commented:
Updated to major version and turned off the formatting rules.

I've also prettified the existing files in this repo, and updated from .prettierrc -> .prettierrc.json so we can easily `require()` and extend the config if need specific plugins for certain projects etc.
